src/pymorize/std_lib/timeaverage.py

```diff
@@ -132,27 +131,46 @@ def _frequency_from_approx_interval(interval: str):
     ValueError
         If the interval cannot be converted to a float.
     """
-    notation = [
-        ("decade", lambda x: f"{x*10}YE" if x else "10YE", 3650),
-        ("year", lambda x: f"{x}YE", 366),
-        ("year", lambda x: f"{x}YE", 365),
-        ("month", lambda x: f"{x}ME", 30),
-        ("day", lambda x: f"{x}D", 1),
-        ("hour", lambda x: f"{x}H", 1 / 24),
-        ("minute", lambda x: f"{x}min", 1.0 / (24 * 60)),
-        ("second", lambda x: f"{x}s", 1.0 / (24 * 60 * 60)),
-        ("millisecond", lambda x: f"{x}ms", 1.0 / (24 * 60 * 60 * 1000)),
-    ]
     try:
         interval = float(interval)
     except ValueError:
         raise ValueError(f"Invalid interval: {interval}")
-    isclose = functools.partial(np.isclose, rtol=1e-3)
-    for name, func, val in notation:
-        if (interval >= val) or isclose(interval, val):
-            value = round(interval / val)
-            value = "" if value == 1 else value
-            return func(value)
+    ref = pd.Timestamp("1970-01-01")
```

Comment:
Why is 1970 taken as a reference year?

Reply:
It is commonly called ‘unix epoch’ or POSIX time. No particular reason to use it. Any non-leap year will do.

Reply:
Haha, alright. Let's leave it as is. If for some reason this ends up biting us, the users can raise an issue. I would at least put a comment next to it, though:

```suggestion
    # NOTE: A reference date is needed to calculate
    #       datetime diffs. We use the Unix epoch as
    #       an arbitrary reference date stamp:
    ref = pd.Timestamp("1970-01-01")
```